Include deleted/Trashed messages in quota calculations.

This patch is no longer maintained.

This patch is distributed under the same terms as qmail-ldap.

Author: John Morrissey <jwm@horde.net>

--- qmail-1.03.stock/Makefile	Tue May 20 19:30:32 2003
+++ qmail-1.03/Makefile	Tue May 20 19:31:48 2003
@@ -16,7 +16,10 @@
 # from a specified account to another (swiss bigbrother law)
 # -DVERIFY_RCPT to verify SMTP RCPT commands against the LDAP directory and
 # alias .qmail files
-LDAPFLAGS=-DDASH_EXT -DEXTERNAL_TODO -DVERIFY_RCPT -DBIGBROTHER
+# -DQUOTA_INCLUDE_TRASH to count messages in the Trash folder when
+# calculating quotas. Normally, messages in the Trash folder are not counted
+# against a user's quota.
+LDAPFLAGS=-DDASH_EXT -DEXTERNAL_TODO -DVERIFY_RCPT -DBIGBROTHER -DQUOTA_INCLUDE_TRASH
 
 
 # SMTP Authentication
@@ -1048,7 +1051,7 @@
 compile maildir++.c maildir++.h readwrite.h stralloc.h error.h str.h \
 open.h substdio.h getln.h error.h strerr.h fmt.h scan.h now.h seek.h \
 sig.h direntry.h
-	./compile maildir++.c
+	./compile $(LDAPFLAGS) maildir++.c
 
 maildir2mbox: \
 load maildir2mbox.o maildir.o prioq.o now.o myctime.o gfrom.o lock.a \
--- qmail-1.03.stock/maildir++.c	Tue May 20 19:28:49 2003
+++ qmail-1.03/maildir++.c	Tue May 20 19:41:25 2003
@@ -386,7 +386,10 @@
 		if ( dp->d_name[0] == '.'
 				&& dp->d_name[1] != '\0' 
 				&& dp->d_name[1] != '.'
-				&& str_diff(".Trash", dp->d_name) ) {
+#ifndef QUOTA_INCLUDE_TRASH
+				&& str_diff(".Trash", dp->d_name)
+#endif
+		) {
 			
 			path.len = plen;
 			if (!stralloc_cats(&path, dp->d_name)) temp_nomem();
@@ -524,7 +527,10 @@
 		if (dp->d_name[0] == '.'
 				&& dp->d_name[1] != '\0'
 				&& dp->d_name[1] != '.'
-				&& str_diff(".Trash", dp->d_name)) {
+#ifndef QUOTA_INCLUDE_TRASH
+				&& str_diff(".Trash", dp->d_name)
+#endif
+		) {
 
 			path.len = slen;
 			if (!stralloc_cats(&path, dp->d_name)) temp_nomem();
@@ -625,6 +631,7 @@
 		while (dirp && (dp = readdir(dirp)) != 0) {
 			f = dp->d_name;
 			if ( *f == '.' ) continue; /* ignore all dot-files */
+#ifndef QUOTA_INCLUDE_TRASH
 			while(*f) {
 				if (*f != ':' || f[1] != '2' || f[2] != ',') {
 					f++;
@@ -637,6 +644,7 @@
 				}
 			}
 			if (*f == 'T') continue;
+#endif
 			/* get the file size */
 			if(get_file_size(dp->d_name, &filest) == 0) {
 				q->count++;
